### PR TITLE
Change Gallery Item to Media Resource

### LINKED ISSUE #84: Rename GalleryItem to MediaResource
We're doing away with the name 'Gallery' anyway, so it doesn't make sense to have a 'GalleryItem'.

### PR DESCRIPTION
**Pull request by audsc**
Media Resource avoids confusion with Profile Items.
Closes #84

### CODE REVIEW

File: km_api/know_me/admin.py [hunk 1]
```diff
@@ -7,10 +7,10 @@
...
 
-@admin.register(models.GalleryItem)
-class GalleryItemAdmin(admin.ModelAdmin):
+@admin.register(models.MediaResource)
+class MediaResourceAdmin(admin.ModelAdmin):
     """
-    Admin for the ``GalleryItem`` model.
+    Admin for the ``MediaResource`` model.
     """
     fields = ('name', 'profile', 'resource')
```
Comment (cdriehuys) [inline km_api/know_me/admin.py]: We changed `resource` to `file` right?

File: km_api/know_me/tests/serializers/test_profile_item_serializer.py [hunk 2]
```diff
@@ -27,27 +30,27 @@ def test_create(gallery_item_factory, profile_row_factory, serializer_context):
...
 
-def test_create_other_user_gallery_item(
-        gallery_item_factory,
+def test_create_other_user_media_resource(
+        media_resource_factory,
         profile_factory,
         serializer_context):
     """
-    Users should not be able to attach a gallery item from a different
+    Users should not be able to attach a gallery resource from a different
```
Comment (cdriehuys) [inline km_api/know_me/tests/serializers/test_profile_item_serializer.py]: Minor nitpick, but this probably shouldn't reference "gallery" at all.

File: km_api/know_me/tests/serializers/test_profile_item_serializer.py [hunk 3]
```diff
@@ -27,27 +30,27 @@ def test_create(gallery_item_factory, profile_row_factory, serializer_context):
...
     profile = profile_factory()
 
     serializer_context['profile'] = profile
 
     data = {
-        'gallery_item': gallery_item.pk,
+        'media_resource': media_resource.pk,
         'name': 'My Profile Item',
-        'text': 'I tried to attach a gallery item from another profile.',
+        'text': 'I tried to attach a gallery resource from another profile.',
```
Comment (cdriehuys) [inline km_api/know_me/tests/serializers/test_profile_item_serializer.py]: Same issue: It doesn't really make sense to reference gallery at all. It is just sample text, so I could be persuaded against the change, but I think it makes sense to change it.

File: km_api/know_me/admin.py (hunk 1 above)
Comment (audsc) [inline km_api/know_me/admin.py]: Correct, will update.

File: km_api/know_me/tests/serializers/test_profile_item_serializer.py (hunk 3 above)
Comment (audsc) [inline km_api/know_me/tests/serializers/test_profile_item_serializer.py]: Nice catch, I agree it doesn't make sense to reference gallery.

File: docs/API/know-me-gallery.rst [hunk 4]
```diff
@@ -2,73 +2,73 @@
...
+    :>json string name: The name the media resource was created with.
+    :>json string resource: The URL of the file attached to the media resource.
 
-    :status 201: The gallery item was succesfully created.
+    :status 201: The media resource was succesfully created.
     :status 400: Invalid request. Check the response data for details.
     :status 404: There is no profile with the given ``id`` accessible to the requesting user.
 
 
 -----------------
```
Comment (cdriehuys) [inline docs/API/know-me-gallery.rst]: Ideally this header would be extended to match the length of the title. I'm not actually sure if this matters in the ReStructured Text spec.

File: docs/API/know-me-gallery.rst [hunk 5]
```diff
@@ -2,73 +2,73 @@
...
-Gallery Item View
+Media Resource View
 -----------------
 
-This endpoint allows for retrieving and updating a specific gallery item's information.
+This endpoint allows for retrieving and updating a specific media resource's information.
 
 .. http:get:: /know-me/gallery-items/(int:id)/
```
Comment (cdriehuys) [inline docs/API/know-me-gallery.rst]: This endpoint should be changed to match the new one: `/know-me/media-resources/`.

File: docs/API/know-me-gallery.rst [hunk 6]
```diff
@@ -2,73 +2,73 @@
...
+    :>json string url: The URL of the media resource's detail view.
+    :>json string name: The name of the media resource.
+    :>json string resource: The URL of the file attached to the media resource.
 
-    :status 200: The gallery item's information was succesfully retrieved.
-    :status 404: There is no gallery item with the given ``id`` accessible to the requesting user.
+    :status 200: The media resource's information was succesfully retrieved.
+    :status 404: There is no media resource with the given ``id`` accessible to the requesting user.
 
 .. http:patch:: /know-me/gallery-items/(int:id)/
```
Comment (cdriehuys) [inline docs/API/know-me-gallery.rst]: As above, the endpoint needs to be updated.

**Pull request merged**